tasks/comment/InfracostComment/index.ts

```diff
@@ -0,0 +1,134 @@
+import taskLib = require('azure-pipelines-task-lib/task');
+import compost from '@infracost/compost';
+import { PostBehavior, TargetType } from '@infracost/compost';
+import { promises as fsPromises } from 'fs';
+
+/**
+ * generateOutput() creates a formatted comment body and saves it to a temporary
+ * file. Returns the file path.
+ *
+ * @param path - path input
+ * @param format - comment format
+ * @param behavior - post behavior, used by `infracost output` command
+ * @return the saved file path
+ */
+function generateOutput(path: string, format: string, behavior: string): string {
+  const outputFile = '/tmp/infracost-comment.md';
+
+  let pathList: string[] = [path];
+  try {
+    pathList = JSON.parse(path);
+  } catch { }
+  const pathArgs = pathList.map(path => `--path ${path}`);
+
+  taskLib.setVariable('INFRACOST_CI_POST_CONDITION', behavior);
+
+  const result = taskLib.execSync(
+    'infracost',
+    [
+      'output',
+      ...pathArgs,
+      `--format ${format}`,
+      `--out-file ${outputFile}`,
+      '--show-skipped',
+    ].join(' ')
+  );
+
+  return result.code === 0 ? outputFile : '';
+}
+
+/**
+ * formatOutput() returns a generated comment body updated with additional notes.
+ *
+ * @param output - comment body
+ * @param format - comment format
+ * @param behavior - post behavior, used by `infracost output` command
+ * @return formatted coment
+ */
+function formatOutput(output: string, targetType: string, behavior: string): string {
+  let note = '';
+
+  if (targetType !== 'commit') {
+    switch (behavior) {
+      case ('update'):
+        note = 'This comment will be updated when the cost estimate changes.';
+        break;
+      case ('delete-and-new'):
+        note = 'This comment will be replaced when the cost estimate changes.';
+        break;
+    }
+  }
+
+  if (note !== '') {
+    output += `\n${note}\n\n`;
+  }
+
+  const link = (value: string, label: string) : string => {
+    return `<a href=\"https://www.infracost.io/feedback/submit/?value=${value}\" rel=\"noopener noreferrer\" target=\"_blank\">${label}</a>`;
+  };
```

Comment:
not sure why we need this function in the scope of `formatOutput` let's put it at the global scope and just define as other functions have been done.

Reply:
Added a `feedbackLink` function 👍 https://github.com/infracost/infracost-azure-devops/pull/11/files#diff-54b9b23a548b3212b9c04189bc051beb211ffc6917ea5e1c82d4523e90ab331eR46-R55